### DISCUSSION TITLE
Installed Homebrew, Seems to Work, but Permissions Errors when Running Installed Apps

### DISCUSSION THREAD

**Discussion by johntdavis84** [Homebrew/discussions#3480] [Getting started] [answered]
Hello, everyone. I've installed Homebrew on a Mac Studio running the latest version of Mac OS, and it works well enough for me to download and install packages. I've grabbed `nano` and `nmap` without a problem.

However, packages that have config files seem to run into what looks like a permissions error when I run them.
Here's what happens when I try to run `micro`:
```
% micro
Error creating configuration directory: mkdir /Users/johndavis/.config/micro: permission denied

Press enter to continue 
open /Users/johndavis/.config/micro/init.lua: permission denied

Press enter to continue
```

I searched the discussions here, and it looks like everyone is having permissions errors with actual `brew` commands, or installing `brew`, but I didn't see anything related specifically to installed packages that don't seem to have proper missions to write to my home directory's `.config/` tree.

Clearly, I've done something wrong, but I'm not sure what. I'd appreciate any advice.

**Comment by JohnMeyerhoff** [comment, marked as answer]
I am assuming this is not an issue on your path and might be resolved by creating the folder structure required by the tool and setting ownership to your account recursively. Could you check, weather the directory for micro actually exists already? And if it does, which permissions does it have?

**Reply by johntdavis84** [reply]
Thanks for your reply. :)

Here's my path:
```
% echo $PATH
/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

However, you made me realize I hadn't checked the permissions, and there is an issue that I don't understand. The entire .config directory is owned by **root**.

That does not seem right.

**EDIT:** I used `chown` to fix the permissions on the `.config` folder, and now everything is fine. Thanks again!

**Reply by JohnMeyerhoff** [reply]
Some Tools will simply not run if you allow them root priviliges, so just prepending sudo to the call is not going to fix it. You have to use whoami to get your username and just `chown` the directory recursively :)

**Reply by johntdavis84** [reply]
> Some Tools will simply not run if you allow them root priviliges, so just prepending sudo to the call is not going to fix it. You have to use whoami to get your username and just `chown` the directory recursively :)

That wasn't so bad to fix, though. Not compared to some things that I've broken/have randomly broken themselves on my servers.

Thanks again for your help. I'll mark this one as closed/solved/whatever that's called here. 👍🏻
